### PR TITLE
[READY] SSticker 

### PR DESCRIPTION
**Pull request by AffectedArc07**
**What does this PR do:**
This PR kills off the ticker process, as well as the gameticker, and turns them both into `SSticker`. This is another piece in my crusade of killing off the GoonPS. Tested acceptable roundstart, not enough players, round end, and a few other things, a testmerge should probably be done but this works.

You're also no-longer able to start the server until all initialisations have been completed, but thats needed at this point to avoid really breaking things.

**Changelog:**
:cl: AffectedArc07
add: SSticker
tweak: You can no longer force-start a round unless its fully initialised. 
/:cl:

### CODE REVIEW

**Comment by taukausanake** [comment]
_Yay stickers!_

File: code/controllers/subsystem/ticker.dm [hunk 1]
```diff
@@ -364,13 +404,13 @@ var/round_start_time = 0
 				qdel(player)
 
 
-/datum/controller/gameticker/proc/collect_minds()
+/datum/controller/subsystem/ticker/proc/collect_minds()
 	for(var/mob/living/player in GLOB.player_list)
 		if(player.mind)
-			ticker.minds += player.mind
+			SSticker.minds += player.mind
```
Comment (Fox-McCloud) [inline code/controllers/subsystem/ticker.dm]: why reference the `SSticker`, when this....is the `SSticker`?

Comment (AffectedArc07) [reply code/controllers/subsystem/ticker.dm]: Why was it referenced like this before?

In all seriousness after you have reviewed this to shreds, I will fix this

File: code/controllers/subsystem/ticker.dm [hunk 2]
```diff
@@ -1,55 +1,64 @@
...
+		lastTickerTimeDuration = (currentTime - (lastTickerTime - TICKS_IN_DAY)) / TICKS_IN_SECOND
+	else
+		lastTickerTimeDuration = (currentTime - lastTickerTime) / TICKS_IN_SECOND
+	lastTickerTime = currentTime
+	
+	// Startup loop
+	if(current_state == GAME_STATE_PREGAME)
 		pregame_timeleft = config.pregame_timestart
 		to_chat(world, "<B><FONT color='blue'>Welcome to the pre-game lobby!</FONT></B>")
 		to_chat(world, "Please, setup your character and select ready. Game will start in [pregame_timeleft] seconds")
```
Comment (Fox-McCloud) [inline code/controllers/subsystem/ticker.dm]: I have a concern about the sleep 2 lines down.

**Comment by Fox-McCloud** [comment]
this may need a bit of additional refactoring to bring us in line with TG's ticker. It doesn't ahve to be 1:1, but getting rid of some of those sleeps would probably be for the best.

**Comment by AffectedArc07** [comment]
Broken as fuck

**Comment by AffectedArc07** [comment]
Time to figure out how TG's ticker works

**Comment by AffectedArc07** [comment]
Has now been tested with someone joining midround, and doesnt have that awful `sleep` in it.

File: code/controllers/subsystem/ticker.dm [hunk 3]
```diff
@@ -1,73 +1,114 @@
...
-		pregame_timeleft = config.pregame_timestart
-		to_chat(world, "<B><FONT color='blue'>Welcome to the pre-game lobby!</FONT></B>")
-		to_chat(world, "Please, setup your character and select ready. Game will start in [pregame_timeleft] seconds")
-		while(current_state == GAME_STATE_PREGAME)
-			sleep(10)
-			if(going)
-				pregame_timeleft--
-
-			if(pregame_timeleft <= 60 && !tipped)
+	..()
```
Comment (Fox-McCloud) [inline code/controllers/subsystem/ticker.dm]: `return ..()`

**Comment by AffectedArc07** [comment]
Closing and re-opening so maints can use appveyor artefacts to test with

**Comment by Fox-McCloud** [comment]
conflicts.

**Comment by AffectedArc07** [comment]
@Fox-McCloud decconflicted

**Comment by MarcusWasTaken** [comment]
Was supposed to comment here after playing the trial round? Anyway, got some z-axis stuttering and problems (specifically in lockers it were flickering which item was on top) as well as the movement seemed less smooth than yesterday

**Comment by AffectedArc07** [comment]
@MarcusWasTaken those are most likely issues due to this being an older game branch, not our latest one. (The latest fixes arent in this branch)

**Comment by Fox-McCloud** [comment]
To resolve the runtime with air alarms:

change this: https://github.com/ParadiseSS13/Paradise/blob/master/code/datums/cache/cache.dm#L3

to this: `var/list/data = list()`

It always being a list is ok; that's all it ever expects.

The only reason this was never a problem is because how the ticker initialize for years. The top `if` of this segment, literally never ran, so it never could runtime, before: https://github.com/ParadiseSS13/Paradise/blob/master/code/datums/cache/air_alarm.dm#L14-L22

**Comment by AffectedArc07** [comment]
Well with https://github.com/ParadiseSS13/Paradise/pull/11523 merged, I am going to merge master locally, and test if it still runtimes, if not this should be good to go

**Comment by AffectedArc07** [comment]
Ready for merge

**Pull request merged**